_overviews/FAQ/initialization-order.md

```diff
@@ -36,129 +36,125 @@ Let's observe the initialization order through the Scala REPL:
 Only when we get to the constructor of `C` are both `x1` and `x2` initialized. Therefore, constructors of `A` and `B` risk running into `NullPointerException`s.
 
 ## Explanation
-A 'strict' or 'eager' val is one which is not marked lazy.
 
-In the absence of "early definitions" (see below), initialization of strict vals is done in the following order.
+A "strict" or "eager" val is one which is not marked lazy.
+Initialization of strict vals is done in the following order:
 
 1. Superclasses are fully initialized before subclasses.
 2. Otherwise, in declaration order.
 
-Naturally when a val is overridden, it is not initialized more than once.  So though x2 in the above example is seemingly defined at every point, this is not the case: an overridden val will appear to be null during the construction of superclasses, as will an abstract val.
+When a `val` is overridden, in fact its accessor method (the "getter") is overridden.
+So the access to `x2` in class `A` in fact invokes the overridden getter in class `C` which reads the underlying field `C.x2`.
+This field is not yet initialized during the construction of `A`.
 
-There is a compiler flag which can be useful for identifying this situation:
+## Mitigation
 
-**-Xcheckinit**: Add runtime check to field accessors.
+The [`-Ysafe-init` compiler flag](https://docs.scala-lang.org/scala3/reference/other-new-features/safe-initialization.html) in Scala 3 enables compiler warnings for accesses to uninitialized fields:
```

Comment:
suggest mentioning here that the flag is experimental, like all `-Y` flags

Reply:
That's like saying use it, but don't like _use_ it. How about instead of experimental, we start saying "cutting-edge". Also, experimental has the narrow meaning for dotty that you need a flag and an experimental compiler to use the feature.

Reply:
it will also be `-Wsafe-init` from 3.5